src/WindowsUtils/Debugger.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WindowsUtils/Debugger.h"
+
+#include "OrbitBase/GetLastError.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/StringConversion.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "WindowsUtils/CreateProcess.h"
+#include "WindowsUtils/SafeHandle.h"
+
+namespace orbit_windows_utils {
+
+Debugger::Debugger() : process_info_or_error_(ErrorMessage()) {}
+Debugger::~Debugger() { Wait(); }
+
+ErrorMessageOr<ProcessInfo> Debugger::Start(const std::filesystem::path& executable,
+                                            const std::filesystem::path& working_directory,
+                                            std::string_view arguments) {
```

Comment:
Nit: Since this is a "sink-argument" I would prefer a `std::string` given by value.

Reply:
Is it really a sink? I'd rather make the arguments to `DebuggerThread ` references and a string_view.

Reply:
We can't make the arguments to `DebuggerThread` references because they might get out of scope by the time they are used. I prefer the status quo actually, it delays copies until they are absolutely needed. We could imagine "earlying-out" with no copies at all.